### PR TITLE
feat: billable checks in project-related doctypes

### PR DESCRIPTION
**Pull request by vishdha**
TASK ID: https://bloomstack.com/desk#Form/Task/TASK-2020-01168

Depends on : https://github.com/DigiThinkIT/frappe/pull/239
Depends on : https://github.com/DigiThinkIT/bloomstack_core/pull/372

### CODE REVIEW

File: erpnext/projects/doctype/project/project.js [hunk 1]
```diff
@@ -130,6 +150,9 @@ function open_form(frm, doctype, child_doctype, parentfield) {
 		new_child_doc.parent = new_doc.name;
 		new_child_doc.parentfield = parentfield;
 		new_child_doc.parenttype = doctype;
+		if (frm.doc.billable) {
+			new_child_doc.billable = frm.doc.billable;
+		}
```
Comment (Alchez) [inline erpnext/projects/doctype/project/project.js]: Remove the condition

File: erpnext/projects/utils.py [hunk 2]
```diff
@@ -26,3 +27,112 @@ def query_task(doctype, txt, searchfield, start, page_len, filters):
...
+
+
+def get_project_time_logs(project):
+	return frappe.get_all("Timesheet Detail", filters={"project": project.name})
+
+def get_task_time_logs(task):
+	return frappe.get_all("Timesheet Detail", filters={"task": task.name})
+
+@frappe.whitelist()
+def get_linked_documents(doctype, name, docs=None):
```
Comment (Alchez) [inline erpnext/projects/utils.py]: How is this function different from the `get_submitted_linked_docs` function?

**Review by Alchez — CHANGES_REQUESTED** [review]

**Comment by Alchez** [comment]
@vishdha, can you update the PR title here

File: erpnext/projects/doctype/project/project.js [hunk 3]
```diff
@@ -130,6 +150,9 @@ function open_form(frm, doctype, child_doctype, parentfield) {
 		new_child_doc.parent = new_doc.name;
 		new_child_doc.parentfield = parentfield;
 		new_child_doc.parenttype = doctype;
+		if (doctype === "Timesheet") {
```
Comment (Alchez) [inline erpnext/projects/doctype/project/project.js:153]: Which other doctype is calling this function?

Comment (vishdha) [reply erpnext/projects/doctype/project/project.js:153]: Purchase Order, Purchase Receipt and Purchase Invoice

File: erpnext/projects/utils.py [hunk 4]
```diff
@@ -5,6 +5,7 @@
 
 from __future__ import unicode_literals
 import frappe
+from frappe.desk.form.linked_with import get_linked_docs, get_linked_doctypes
```
Comment (Alchez) [inline erpnext/projects/utils.py]: Remove these imports

File: erpnext/public/js/utils.js [hunk 5]
```diff
@@ -653,4 +653,61 @@ $(document).on('app_ready', function() {
...
+	$.each(["Project", "Task", "Project Template", "Project Type"], function (i, doctype) {
+		frappe.ui.form.on(doctype, {
+			billable: (frm) => {
+				let confirm_message = "";
+				frappe.call({
+					method: "frappe.desk.form.linked_with.get_submitted_linked_docs",
+					args: {
+						doctype: frm.doc.doctype,
+						name: frm.doc.name,
+						is_submittable: 0,
```
Comment (Alchez) [inline erpnext/public/js/utils.js]:
```suggestion
						is_submittable: false,
```

**Review by Alchez — CHANGES_REQUESTED** [review]

**Review by Alchez — DISMISSED** [review]

**Review by Alchez — APPROVED** [review]

**Pull request merged**